crates/bevy_render/src/camera/visible_entities.rs

```diff
@@ -24,22 +24,208 @@ impl VisibleEntities {
     }
 }
 
+type LayerMask = u32;
+
+/// An identifier for a rendering layer.
+pub type Layer = u8;
+
+/// Describes which rendering layers an entity belongs to.
+///
+/// Cameras with this component will only render entities with intersecting
+/// layers.
+///
+/// There are 32 layers numbered `0` - [`TOTAL_LAYERS`]. Entities may belong to one or more
+/// layers, or no layer at all.
+///
+/// The [`Default`] instance of `RenderLayers` contains layer `0`, the first layer.
+///
+/// An entity with this component without any layers is invisible.
```

Comment:
I'm still cautious about making this guarantee, since we still seem to require that every entity has a `Visible`, which this overlaps with

Reply:
Yes, it's a bit sketchy but it does help describe the feature. I think this is the best we can do at the moment. I personally plan on using the "invisiblity" of `RenderLayers::none()` so I don't think we should add a warning.

Reply:
Yeah I thought a bit about just replacing Visible with RenderLayers. But I think we need a "global" visible/invisible toggle:

* That way visibility can still be toggled without needing to know what layers the current entity is visible on
* Visibility will likely eventually propagate down a hierarchy, while RenderLayers _might_ not. Its worth discussing at some point, but imo doesn't need to be settled in this pr.
* I want entity visibility to be easily toggle-able in the editor from the "Entity Inspector" (ex: have an "eye" toggle next to the entity name), which would be much harder for arbitrary layers